review/get_review_ids.py

```diff
@@ -0,0 +1,239 @@
+# pylint: disable=import-error
+'''
+The logic for actually grabbing review content is contained here for the
+Review XBlock. This works by having a copy of the actual course a learner
+is interacting with that is hidden from learners. This copied course hosts
+the content we will show as review so the review content displayed has a
+fresh state and can be set to ungraded so it doesn't affect the learners'
+grades. There are two ways review content can be grabbed:
+    On a per problem basis
+    On a unit basis (this would be single view that contains multiple problems)
+'''
+
+import logging
+
+from datetime import datetime
+import json
+import random
+from courseware.models import StudentModule
+from django.conf import settings
+from enrollment.api import get_enrollment, add_enrollment, update_enrollment
+from xmodule.modulestore.django import modulestore
+import crum
+import pytz
+
+from .configuration import REVIEW_COURSE_MAPPING, ENROLLMENT_COURSE_MAPPING, TEMPLATE_URL
+
+# This check exists because if we import lms.djangoapps.instructor.enrollment
+# when it's not in the django INSTALLED_APPS, many python tests running in the
+# CMS environment fail since the tests then expect a model to exist, but it
+# doesn't because it is not in django's INSTALLED_APPS. By verifying we are in
+# the LMS through the check, we avoid these test failures.
+if 'lms.djangoapps.grades.apps.GradesConfig' in settings.INSTALLED_APPS:
+    from lms.djangoapps.course_blocks.api import get_course_blocks
+    from lms.djangoapps.instructor.enrollment import reset_student_attempts
+
+log = logging.getLogger(__name__)
+
+# TODO: Switch to using CourseLocators and/or CourseKeys everywhere
+
+
+def get_problems(num_desired, current_course):
+    '''
+    Looks through all the problems a learner has previously loaded and randomly
+    selects num_desired of them. Also checks if the learner had originally
+    answered it correctly or incorrectly and after how many attempts.
+
+    Parameters:
+        num_desired (int): the number of desired problems to show the learner
+        current_course (CourseLocator): The course the learner is currently in
+
+    Returns a list of num_desired tuples in the form (URL to display, correct, attempts)
+    '''
+    user = crum.get_current_user()
+
+    enroll_user_in_review_course(user, current_course)
+    store = modulestore()
+
+    problem_data = []
+
+    for block_key, state in get_records(user, current_course):
+        if is_valid_problem(store, block_key, state):
+            correct, attempts = get_correctness_and_attempts(state)
+            problem_id = block_key.block_id
+            problem_data.append((problem_id, correct, attempts))
+            delete_state_of_review_problem(user, current_course, problem_id)
+
+    if len(problem_data) < num_desired:
+        return []
+
+    problems_to_show = random.sample(problem_data, num_desired)
+    review_course_id = REVIEW_COURSE_MAPPING[str(current_course)]
+    urls = []
```

Comment:
I would argue that `urls` isn't the best name for this because it contains other things besides urls.